### PR TITLE
feat: Hot reload python files when running locally

### LINKED ISSUE #91: Hot reload python lambdas during dev.

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

This PR does a couple of things.

First it reworks the Dockerfile for the API so that it runs expensive commands first so we can take advantage of caching.

For instance it will install playwright and chromium up front since those are slow and as it stands will always re-run whenever files change.

Secondly it installs the python modules in a different folder then the api code and sets the `PYTHONPATH` env variable to that folder so that we can mount the `api` folder as a volume mounted to `/function/` in the container.

Thirdly it installs `entr` and uses that to watch for changes to all *.py files in the api folder and subdirectories, when those files change it will restart aws-lambda-rie.

Closes #91

# Test instructions | Instructions pour tester la modification

Run the branch in a devcontainer, make a change to the python code, hit the endpoint and see the new code change occur.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -33,6 +33,6 @@
 	// Use 'forwardPorts' to make a list of ports inside the container available locally.
 	// "forwardPorts": [],
 	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "pip3 install --user -r api/requirements.txt && pip3 install --user -r api/requirements_dev.txt && python -m playwright install chromium",
+	"postCreateCommand": "",
```
Comment (patheard) [inline .devcontainer/devcontainer.json:36]: nit - since this is now empty, you can smoke the line.

File: api/Dockerfile [hunk 2]
```diff
@@ -10,23 +10,30 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
...
+
+RUN playwright install chromium
 
-RUN python3.9 -m pip install -r ${FUNCTION_DIR}/requirements.txt --target ${FUNCTION_DIR}
+COPY ./requirements.txt ${FUNCTION_DIR}
 
+
+RUN python3.9 -m pip install -r ${FUNCTION_DIR}/requirements.txt --target /pymodules
+
+RUN apt-get install entr
```
Comment (patheard) [inline api/Dockerfile:29]: Do we need an `apt-get update` here?

**Review by patheard — COMMENTED** [review]
Few minor comments, but it works oh so smooth for me locally.

File: api/Dockerfile (hunk 2 above)
Comment (CalvinRodo) [reply api/Dockerfile:29]: We update on line 20 I'll add entr to that line instead.

**Review by mohamed-cds — APPROVED** [review]
Worked like a charm 🚀 ✅

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (CalvinRodo) [reply .devcontainer/devcontainer.json:36]: done

File: api/Dockerfile (hunk 2 above)
Comment (CalvinRodo) [reply api/Dockerfile:29]: added it to other apt-get install

**Review by patheard — APPROVED** [review]

**Pull request merged**